### DISCUSSION TITLE
Issue making install public

### DISCUSSION THREAD

**Discussion by cornerkilling** [mcguirepr89/BirdNET-Pi#255] [Q&A]
Hi, I recently installed this app and am loving the results. I was hoping to make my install public to share with family. I followed the instructions for working with a No-IP address. All ports are forwarded and everything looks okay in the No-IP dashboard. I also am able to access everything perfectly fine through the local address. 

When accessing the public address I get a page to load but no data is appearing. Just the header and background. Any ideas for troubleshoot this? I thought at first maybe it was just loading slowly but I've let it sit for a while with no change. 
![image](https://user-images.githubusercontent.com/97748126/167310305-e6e005c9-0832-4f91-aeb9-598993b61349.png)

**Comment by cornerkilling** [comment]
> Hi, I recently installed this app and am loving the results. I was hoping to make my install public to share with family. I followed the instructions for working with a No-IP address. All ports are forwarded and everything looks okay in the No-IP dashboard. I also am able to access everything perfectly fine through the local address.
> 
> When accessing the public address I get a page to load but no data is appearing. Just the header and background. Any ideas for troubleshoot this? I thought at first maybe it was just loading slowly but I've let it sit for a while with no change. ![image](https://user-images.githubusercontent.com/97748126/167310305-e6e005c9-0832-4f91-aeb9-598993b61349.png)

I will also mention that the live audio feature does work on the public address.

**Comment by ehpersonal38** [comment]
Hi there! I'll try my best to help you troubleshoot this!

Can you go to {yourdomain}/views.php and screenshot the outcome?

Also can you go to {yourdomain}/index.php, press ctrl + shift + j, and screenshot the console and any error messages?

**Reply by cornerkilling** [reply]
Here is the only error from the index
crbug/1173575, non-JS module files deprecated.
The views page is timing out. I tried pining the Pi and it is still connected okay. 
![image](https://user-images.githubusercontent.com/97748126/167311133-0bee9673-432b-4341-b3c9-c813337fd7df.png)

**Comment by mcguirepr89** [comment]
Hello @cornerkilling and @ehpersonal38 

This sounds like a web browser caching issue. Could you try accessing your installation using an "Incognito" window? or clearing your browser cache to see if that helps?

**Reply by mcguirepr89** [reply]
Now that I can see the domain and have tested myself (which would rule our caching), it's definitely not that.

**Reply by ehpersonal38** [reply]
Yeah this is strange - something on views.php is running an infinite loop, or rather sending a web request that doesn't have a timeout.

**Reply by ehpersonal38** [reply]
Looks like it's working now? Edit: and down again

**Reply by mcguirepr89** [reply]
It finally loaded for me, but there is something going on with the webserver perhaps?

**Reply by cornerkilling** [reply]
Yes, seems to be working now. I power cycled the Pi and then it popped right up. Not sure what took it down though. The pi was still show 5ms pings so it seems to have had a fine connection.

**Comment by mcguirepr89** [comment]
Next place to look is in `caddy`'s log:
```
journalctl --no-hostname -ucaddy -n 100 --no-pager
```

**Reply by mcguirepr89** [reply]
I'm hitting the reverse_proxies without issue https://birdfish.ddns.net/log so the webserver is working mostly???? But something else is up obviously

**Reply by mcguirepr89** [reply]
Wow it's really working very hard at the moment. This looks like it needs to have a few updates to hopefully reduce a little of that load?
```
15:13:10 server.py[402]: Database busy
15:13:10 server.py[402]: Database busy
15:13:10 server.py[402]: Database busy
15:13:10 server.py[402]: 2022-05-08;15:09:09;Sayornis phoebe;Eastern Phoebe;0.96792936;40.1149;-82.9054;0.65;18;1.25;0.0Eastern_Phoebe-0.96792936-2022-05-08-birdnet-15:09:09.mp3
15:13:10 server.py[402]: Database busy
15:13:10 server.py[402]: Database busy
15:13:10 server.py[402]: Database busy
15:13:10 server.py[402]: 2022-05-08;15:09:09;Sayornis phoebe;Eastern Phoebe;0.92420626;40.1149;-82.9054;0.65;18;1.25;0.0Eastern_Phoebe-0.92420626-2022-05-08-birdnet-15:09:09.mp3
15:13:10 server.py[402]: Database busy
15:13:10 server.py[402]: Database busy
15:13:10 server.py[402]: Database busy
15:13:10 server.py[402]: 2022-05-08;15:09:09;Sayornis phoebe;Eastern Phoebe;0.9638277;40.1149;-82.9054;0.65;18;1.25;0.0Eastern_Phoebe-0.9638277-2022-05-08-birdnet-15:09:09.mp3
15:13:10 server.py[402]: Database busy
15:13:10 server.py[402]: Database busy
15:13:10 server.py[402]: Database busy
15:13:10 server.py[402]: 2022-05-08;15:09:09;Sayornis phoebe;Eastern Phoebe;0.9894254;40.1149;-82.9054;0.65;18;1.25;0.0Eastern_Phoebe-0.9894254-2022-05-08-birdnet-15:09:09.mp3
```

**Reply by mcguirepr89** [reply]
That Eastern Phoebe has a lot to say right now...

**Reply by ehpersonal38** [reply]
Haha - something is definitely using a lot of cpu

**Reply by mcguirepr89** [reply]
SO it looks like the page is not loading because the database is locking up from so many requests (looks similar to when the repo got all popular and my installations went down). 
OR `php7.4-fpm.service` crashed for some reason?

@cornerkilling could you please run:

```
sudo systemctl restart php7\*-fpm.service
```

~~if `php7.4-fpm.service` failed, then it would explain not being able to see any php scripts, but being able to stream the live audio and hit the reverse proxies without issue. Also, `caddy` was able to fetch the SSL certificates, so the routing is good for what that is worth at this point.~~

~~I'm leaning towards `php7.4-fpm.service` having messed up for some reason. We'll see whether restarting the service.~~

NEvermind, I forget that the index is now `index.php` . . . so my theory doesn't make sense

**Reply by ehpersonal38** [reply]
What would be generating all those requests though?

**Reply by cornerkilling** [reply]
May be a dumb question but how do I copy or export the log that you said to pull from the web terminal? Or should I run that on the pi itself so I can copy and paste here?

**Reply by mcguirepr89** [reply]
@ehpersonal38 
>What would be generating all those requests though?

I'm not sure I'm aware of which requests you mean?

I'm requesting specific things and hitting a timeout, I'm not seeing requests looping anywhere from my vantage

**Reply by mcguirepr89** [reply]
If you run:
```
journalctl --no-hostname -ucaddy -n 100 --no-pager
```

In the web terminal, you can just copy and paste the results here (If highlighting the text and right clicking to select "Copy" isn't working, try holding down "Alt" as you select the text, or "Shift" if "Alt" doesn't work.

**Reply by ehpersonal38** [reply]
`it looks like the page is not loading because the database is locking up from so many requests`

I mean what is querying the database so much that it causes everything to lock up? - and what is specific to his installation that isn't occuring on anyone else's installation?

and why can he `access everything perfectly fine through the local address` but not through the internet? It's the same server isn't it? 😕

**Reply by cornerkilling** [reply]
```
-- Journal begins at Mon 2022-04-04 10:52:30 EDT, ends at Sun 2022-05-08 15:33:22 EDT. --
May 06 20:18:24 caddy[613]: caddy.AppConfigDir=/var/lib/caddy/.config/caddy
May 06 20:18:24 caddy[613]: caddy.ConfigAutosavePath=/var/lib/caddy/.config/caddy/autosave.json
May 06 20:18:24 caddy[613]: caddy.Version=v2.5.0 h1:eRHzZ4l3X6Ag3kUt8nj5IxATprhqKq/wToP7OHlXWA0=
May 06 20:18:24 caddy[613]: runtime.GOOS=linux
May 06 20:18:24 caddy[613]: runtime.GOARCH=arm64
May 06 20:18:24 caddy[613]: runtime.Compiler=gc
May 06 20:18:24 caddy[613]: runtime.NumCPU=4
May 06 20:18:24 caddy[613]: runtime.GOMAXPROCS=4
May 06 20:18:24 caddy[613]: runtime.Version=go1.18.1
May 06 20:18:24 caddy[613]: os.Getwd=/
May 06 20:18:24 caddy[613]: LANG=en_GB.UTF-8
May 06 20:18:24 caddy[613]: PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
May 06 20:18:24 caddy[613]: NOTIFY_SOCKET=/run/systemd/notify
May 06 20:18:24 caddy[613]: HOME=/var/lib/caddy
May 06 20:18:24 caddy[613]: LOGNAME=caddy
May 06 20:18:24 caddy[613]: USER=caddy
May 06 20:18:24 caddy[613]: INVOCATION_ID=c6f253dbc23e49caac8006e2ae3d529e
May 06 20:18:24 caddy[613]: JOURNAL_STREAM=8:14256
May 06 20:18:24 caddy[613]: {"level":"info","ts":1651882704.193461,"msg":"using provided configuration","config_file":"/etc/caddy/Caddyfile","config_adapter":""}
May 06 20:18:24 caddy[613]: {"level":"warn","ts":1651882704.2314725,"msg":"Caddyfile input is not formatted; run the 'caddy fmt' command to fix inconsistencies","adapter":"caddyfile","file":"/etc/caddy/Caddyfile","line":2}
May 06 20:18:24 caddy[613]: {"level":"info","ts":1651882704.2986655,"logger":"admin","msg":"admin endpoint started","address":"tcp/localhost:2019","enforce_origin":false,"origins":["//localhost:2019","//[::1]:2019","//127.0.0.1:2019"]}
May 06 20:18:24 caddy[613]: {"level":"info","ts":1651882704.2999399,"logger":"http","msg":"server is listening only on the HTTPS port but has no TLS connection policies; adding one to enable TLS","server_name":"srv0","https_port":443}
May 06 20:18:24 caddy[613]: {"level":"info","ts":1651882704.3004532,"logger":"http","msg":"enabling automatic HTTP->HTTPS redirects","server_name":"srv0"}
May 06 20:18:24 caddy[613]: {"level":"warn","ts":1651882704.3007808,"logger":"http","msg":"server is listening only on the HTTP port, so no automatic HTTPS will be applied to this server","server_name":"srv1","http_port":80}
May 06 20:18:24 caddy[613]: {"level":"info","ts":1651882704.2999845,"logger":"tls.cache.maintenance","msg":"started background certificate maintenance","cache":"0x40003d3f10"}
May 06 20:19:03 caddy[613]: {"level":"info","ts":1651882743.3678792,"logger":"http","msg":"enabling automatic TLS certificate management","domains":["birdfish.ddns.net"]}
May 06 20:19:04 caddy[613]: {"level":"info","ts":1651882744.443369,"logger":"tls","msg":"cleaning storage unit","description":"FileStorage:/var/lib/caddy/.local/share/caddy"}
May 06 20:19:04 caddy[613]: {"level":"info","ts":1651882744.4702537,"msg":"autosaved config (load with --resume flag)","file":"/var/lib/caddy/.config/caddy/autosave.json"}
May 06 20:19:04 systemd[1]: Started Caddy.
May 06 20:19:04 caddy[613]: {"level":"info","ts":1651882744.5112438,"msg":"serving initial configuration"}
May 06 20:19:04 caddy[613]: {"level":"info","ts":1651882744.511887,"logger":"tls","msg":"finished cleaning storage units"}
May 06 20:26:03 caddy[613]: {"level":"error","ts":1651883163.083049,"logger":"http.handlers.reverse_proxy","msg":"aborting with incomplete response","error":"context canceled"}
May 07 20:19:05 caddy[613]: {"level":"info","ts":1651969145.113569,"logger":"tls","msg":"cleaning storage unit","description":"FileStorage:/var/lib/caddy/.local/share/caddy"}
May 07 20:19:05 caddy[613]: {"level":"info","ts":1651969145.3577147,"logger":"tls","msg":"finished cleaning storage units"}
May 08 14:06:38 caddy[613]: {"level":"error","ts":1652033198.031786,"logger":"http.handlers.reverse_proxy","msg":"aborting with incomplete response","error":"context canceled"}
May 08 14:13:13 systemd[1]: Reloading Caddy.
May 08 14:13:13 caddy[2123196]: {"level":"info","ts":1652033593.840881,"msg":"using provided configuration","config_file":"/etc/caddy/Caddyfile","config_adapter":""}
May 08 14:13:13 caddy[2123196]: {"level":"warn","ts":1652033593.848763,"msg":"Caddyfile input is not formatted; run the 'caddy fmt' command to fix inconsistencies","adapter":"caddyfile","file":"/etc/caddy/Caddyfile","line":2}
May 08 14:13:13 caddy[2123196]: {"level":"info","ts":1652033593.8496552,"msg":"using provided configuration","config_file":"/etc/caddy/Caddyfile","config_adapter":""}
May 08 14:13:13 caddy[2123196]: {"level":"warn","ts":1652033593.8544474,"msg":"Caddyfile input is not formatted; run the 'caddy fmt' command to fix inconsistencies","adapter":"caddyfile","file":"/etc/caddy/Caddyfile","line":2}
May 08 14:13:13 caddy[613]: {"level":"info","ts":1652033593.858863,"logger":"admin.api","msg":"received request","method":"POST","host":"localhost:2019","uri":"/load","remote_ip":"127.0.0.1","remote_port":"52360","headers":{"Accept-Encoding":["gzip"],"Content-Length":["6246"],"Content-Type":["application/json"],"Origin":["http://localhost:2019"],"User-Agent":["Go-http-client/1.1"]}}
May 08 14:13:13 caddy[613]: {"level":"info","ts":1652033593.872754,"logger":"admin","msg":"admin endpoint started","address":"tcp/localhost:2019","enforce_origin":false,"origins":["//localhost:2019","//[::1]:2019","//127.0.0.1:2019"]}
May 08 14:13:13 caddy[613]: {"level":"info","ts":1652033593.8738658,"logger":"tls.cache.maintenance","msg":"started background certificate maintenance","cache":"0x40003ce000"}
May 08 14:13:13 caddy[613]: {"level":"info","ts":1652033593.873941,"logger":"http","msg":"server is listening only on the HTTPS port but has no TLS connection policies; adding one to enable TLS","server_name":"srv0","https_port":443}
May 08 14:13:13 caddy[613]: {"level":"info","ts":1652033593.8739789,"logger":"http","msg":"enabling automatic HTTP->HTTPS redirects","server_name":"srv0"}
May 08 14:13:13 caddy[613]: {"level":"warn","ts":1652033593.874018,"logger":"http","msg":"server is listening only on the HTTP port, so no automatic HTTPS will be applied to this server","server_name":"srv1","http_port":80}
May 08 14:13:23 caddy[613]: {"level":"error","ts":1652033603.8736622,"logger":"admin","msg":"stopping current admin endpoint","error":"shutting down admin server: context deadline exceeded"}
May 08 14:13:33 caddy[613]: {"level":"info","ts":1652033613.2353456,"logger":"tls","msg":"cleaning storage unit","description":"FileStorage:/var/lib/caddy/.local/share/caddy"}
May 08 14:13:33 caddy[613]: {"level":"info","ts":1652033613.2355468,"logger":"http","msg":"enabling automatic TLS certificate management","domains":["birdfish.ddns.net"]}
May 08 14:13:33 caddy[613]: {"level":"info","ts":1652033613.2474475,"logger":"tls","msg":"finished cleaning storage units"}
May 08 14:13:33 caddy[613]: {"level":"info","ts":1652033613.2548473,"logger":"tls.cache.maintenance","msg":"stopped background certificate maintenance","cache":"0x40003d3f10"}
May 08 14:13:33 caddy[613]: {"level":"info","ts":1652033613.2562006,"msg":"autosaved config (load with --resume flag)","file":"/var/lib/caddy/.config/caddy/autosave.json"}
May 08 14:13:33 caddy[613]: {"level":"info","ts":1652033613.2564414,"logger":"admin.api","msg":"load complete"}
May 08 14:13:33 systemd[1]: Reloaded Caddy.
May 08 14:48:46 caddy[613]: {"level":"error","ts":1652035726.048418,"logger":"http.handlers.reverse_proxy","msg":"aborting with incomplete response","error":"write tcp [2600:1700:8820:9c40:3bda:9f94:faef:d10a]:80->[2600:1700:8820:9c40:fc19:84c1:6902:466c]:62585: write: broken pipe"}
May 08 14:50:50 caddy[613]: {"level":"error","ts":1652035850.8036044,"logger":"http.handlers.reverse_proxy","msg":"aborting with incomplete response","error":"http2: stream closed"}
-- Boot 9661becf53af43bf8a09e26ccd35c7ab --
May 08 14:56:30 systemd[1]: Starting Caddy...
May 08 14:56:35 caddy[595]: caddy.HomeDir=/var/lib/caddy
May 08 14:56:35 caddy[595]: caddy.AppDataDir=/var/lib/caddy/.local/share/caddy
May 08 14:56:35 caddy[595]: caddy.AppConfigDir=/var/lib/caddy/.config/caddy
May 08 14:56:35 caddy[595]: caddy.ConfigAutosavePath=/var/lib/caddy/.config/caddy/autosave.json
May 08 14:56:35 caddy[595]: caddy.Version=v2.5.0 h1:eRHzZ4l3X6Ag3kUt8nj5IxATprhqKq/wToP7OHlXWA0=
May 08 14:56:35 caddy[595]: runtime.GOOS=linux
May 08 14:56:35 caddy[595]: runtime.GOARCH=arm64
May 08 14:56:35 caddy[595]: runtime.Compiler=gc
May 08 14:56:35 caddy[595]: runtime.NumCPU=4
May 08 14:56:35 caddy[595]: runtime.GOMAXPROCS=4
May 08 14:56:35 caddy[595]: runtime.Version=go1.18.1
May 08 14:56:35 caddy[595]: os.Getwd=/
May 08 14:56:35 caddy[595]: LANG=en_GB.UTF-8
May 08 14:56:35 caddy[595]: PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
May 08 14:56:35 caddy[595]: NOTIFY_SOCKET=/run/systemd/notify
May 08 14:56:35 caddy[595]: HOME=/var/lib/caddy
May 08 14:56:35 caddy[595]: LOGNAME=caddy
May 08 14:56:35 caddy[595]: USER=caddy
May 08 14:56:35 caddy[595]: INVOCATION_ID=440694e314c144f1865c5751b68e5663
May 08 14:56:35 caddy[595]: JOURNAL_STREAM=8:13098
May 08 14:56:35 caddy[595]: {"level":"info","ts":1652036195.9022994,"msg":"using provided configuration","config_file":"/etc/caddy/Caddyfile","config_adapter":""}
May 08 14:56:35 caddy[595]: {"level":"warn","ts":1652036195.9165747,"msg":"Caddyfile input is not formatted; run the 'caddy fmt' command to fix inconsistencies","adapter":"caddyfile","file":"/etc/caddy/Caddyfile","line":2}
May 08 14:56:35 caddy[595]: {"level":"info","ts":1652036195.95957,"logger":"admin","msg":"admin endpoint started","address":"tcp/localhost:2019","enforce_origin":false,"origins":["//localhost:2019","//[::1]:2019","//127.0.0.1:2019"]}
May 08 14:56:35 caddy[595]: {"level":"info","ts":1652036195.9609177,"logger":"tls.cache.maintenance","msg":"started background certificate maintenance","cache":"0x400044d260"}
May 08 14:56:35 caddy[595]: {"level":"info","ts":1652036195.9612658,"logger":"http","msg":"server is listening only on the HTTPS port but has no TLS connection policies; adding one to enable TLS","server_name":"srv0","https_port":443}
May 08 14:56:35 caddy[595]: {"level":"info","ts":1652036195.9613323,"logger":"http","msg":"enabling automatic HTTP->HTTPS redirects","server_name":"srv0"}
May 08 14:56:35 caddy[595]: {"level":"warn","ts":1652036195.9614968,"logger":"http","msg":"server is listening only on the HTTP port, so no automatic HTTPS will be applied to this server","server_name":"srv1","http_port":80}
-- Boot 9044368dc2cb458ab4fa420a99e18441 --
May 08 14:57:06 caddy[613]: {"level":"error","ts":1652036180.2092836,"logger":"http.handlers.reverse_proxy","msg":"aborting with incomplete response","error":"client disconnected"}
May 08 14:57:06 caddy[613]: {"level":"error","ts":1652036200.4932258,"logger":"http.handlers.reverse_proxy","msg":"aborting with incomplete response","error":"client disconnected"}
May 08 15:01:04 caddy[613]: {"level":"error","ts":1652036444.9654617,"logger":"http.log.error","msg":"dialing backend: dial unix /run/php/php7.4-fpm.sock: i/o timeout","request":{"remote_ip":"192.168.1.254","remote_port":"443","proto":"HTTP/2.0","method":"GET","host":"birdfish.ddns.net","uri":"/views.php","headers":{"Sec-Ch-Ua-Platform":["\"Windows\""],"Dnt":["1"],"Sec-Fetch-Site":["none"],"Sec-Fetch-User":["?1"],"Accept-Encoding":["gzip, deflate, br"],"Sec-Ch-Ua-Mobile":["?0"],"Accept-Language":["en-US,en;q=0.9"],"Sec-Fetch-Mode":["navigate"],"Accept":["text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9"],"Sec-Fetch-Dest":["document"],"Upgrade-Insecure-Requests":["1"],"Sec-Ch-Ua":["\" Not A;Brand\";v=\"99\", \"Chromium\";v=\"100\", \"Google Chrome\";v=\"100\""],"User-Agent":["Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.127 Safari/537.36"],"Authorization":[]},"tls":{"resumed":false,"version":772,"cipher_suite":4867,"proto":"h2","server_name":"birdfish.ddns.net"}},"duration":51.206691146,"status":502,"err_id":"z8ycvdgh3","err_trace":"reverseproxy.statusError (reverseproxy.go:1166)"}
-- Boot 9661becf53af43bf8a09e26ccd35c7ab --
May 08 15:08:39 caddy[595]: {"level":"info","ts":1652036919.4059317,"logger":"tls","msg":"cleaning storage unit","description":"FileStorage:/var/lib/caddy/.local/share/caddy"}
May 08 15:08:39 caddy[595]: {"level":"info","ts":1652036919.5726821,"logger":"http","msg":"enabling automatic TLS certificate management","domains":["birdfish.ddns.net"]}
May 08 15:08:39 caddy[595]: {"level":"info","ts":1652036919.770343,"logger":"tls","msg":"finished cleaning storage units"}
May 08 15:08:39 caddy[595]: {"level":"info","ts":1652036919.887468,"msg":"autosaved config (load with --resume flag)","file":"/var/lib/caddy/.config/caddy/autosave.json"}
May 08 15:08:39 systemd[1]: Started Caddy.
May 08 15:08:39 caddy[595]: {"level":"info","ts":1652036919.909547,"msg":"serving initial configuration"}
May 08 15:13:05 caddy[595]: {"level":"error","ts":1652037185.2368536,"logger":"http.handlers.reverse_proxy","msg":"aborting with incomplete response","error":"http2: stream closed"}
May 08 15:13:05 caddy[595]: {"level":"error","ts":1652037185.5069566,"logger":"http.handlers.reverse_proxy","msg":"aborting with incomplete response","error":"http2: stream closed"}
May 08 15:21:32 caddy[595]: {"level":"error","ts":1652037692.7315857,"logger":"http.handlers.reverse_proxy","msg":"aborting with incomplete response","error":"http2: stream closed"}
May 08 15:32:23 caddy[595]: {"level":"error","ts":1652038343.4457352,"logger":"http.handlers.reverse_proxy","msg":"aborting with incomplete response","error":"http2: stream closed"}
May 08 15:32:23 caddy[595]: {"level":"error","ts":1652038343.4640784,"logger":"http.handlers.reverse_proxy","msg":"aborting with incomplete response","error":"http2: stream closed"}
May 08 15:32:23 caddy[595]: {"level":"error","ts":1652038343.4713876,"logger":"http.handlers.reverse_proxy","msg":"aborting with incomplete response","error":"http2: stream closed"}
May 08 15:32:23 caddy[595]: {"level":"error","ts":1652038343.4764063,"logger":"http.handlers.reverse_proxy","msg":"aborting with incomplete response","error":"http2: stream closed"}
```

**Reply by cornerkilling** [reply]
> `it looks like the page is not loading because the database is locking up from so many requests`
> 
> I mean what is querying the database so much that it causes everything to lock up? - and what is specific to his installation that isn't occuring on anyone else's installation?
> 
> and why can he `access everything perfectly fine through the local address` but not through the internet? It's the same server isn't it? 😕

Should mention that when it has been going down lately while troubleshooting, the local address also fails. I can still access the Pi through VNC to run these commands for the logs.

**Reply by mcguirepr89** [reply]
Now I'm getting better responses:
![image](https://user-images.githubusercontent.com/60325264/167312998-bcdb027d-62f9-4b1d-a8bf-eef5c7e755e5.png)

**Reply by mcguirepr89** [reply]
So strange -- I'm wondering is there is something in `index.php` that is broken? I'm able to go to https://birdfish.ddns.net/views.php and see the overview, but I'm getting nothing visiting /index.php :thinking:

**Reply by ehpersonal38** [reply]
views.php doesn't load for me

**Reply by cornerkilling** [reply]
Yea both local and public addresses are not loading now

**Reply by ehpersonal38** [reply]
Just a thought, could you run `bash $HOME/BirdNET-Pi/scripts/stop_core_services.sh` and see if the webserver is still locking up? Then maybe we could turn on each service one by one to see where the problem lies...

**Reply by cornerkilling** [reply]
Okay so I had to power cycle the pi again to get back into the web view. I ran that script and activated the services one by one. Working better except View Log on the public address is not loading anything. I occasionally get time out issues still on the public address. Not sure what changed since I had power cycled once before and did not have this happen.

**Reply by ehpersonal38** [reply]
Yes I still don't know the root cause... did you mention what raspberry Pi model you're using?

Also with the stop all services above ^^ I was thinking: stop core services, see if web interface is loading any better, if so then we can turn each service on one by one, check webpage again, until we hit the one that starts locking things up.

**Reply by cornerkilling** [reply]
It's a model 4b

**Reply by mcguirepr89** [reply]
It looks like it's working now -- can you confirm if it is working for you guys as well?

Be sure to use an incognito browser to not load any old pages.
![image](https://user-images.githubusercontent.com/60325264/167317097-68cd4c38-82bb-45f4-8a2b-178280d17db3.png)

**Reply by ehpersonal38** [reply]
> It looks like it's working now -- can you confirm if it is working for you guys as well?

![chrome_nKUupqLyUb](https://user-images.githubusercontent.com/103586016/167318591-923b91bf-8ec9-4711-bfee-2bc9d693114e.png)

Unfortunately not. You can see my tab in the top left still loading, and accessing it from Browserling (an online virtual machine) shows the same. The /log and chart images load quickly though, definitely a php problem.

**Reply by cornerkilling** [reply]
Should I try a fresh install? The only thing I can think of that might be unique about this install is I have PIOS installed on a usb ssd instead of an sd card.

**Comment by cornerkilling** [comment]
Getting a few errors on my end. 
/overview.php?ajax_detections=true&previous_detection_identifier=undefined:1          Failed to load resource: the server responded with a status of 500 (Internal Server Error)
/overview.php?ajax_left_chart=true:1          Failed to load resource: the server responded with a status of 500 (Internal Server Error)

**Reply by ehpersonal38** [reply]
Those are normal afaik

**Reply by cornerkilling** [reply]
Okay, I checked the errors because it was timing out again.

**Comment by mcguirepr89** [comment]
That makes sense -- as we've gone through the logs, it has become apparent
that the issue lives somewhere on your LAN. The system itself is working
without issue and the trouble we've been able to observe has had to do with
packets getting lost during requests, causing timeouts. Having resources
fight over the packet ports (80 and 443) will definitely cause this to
happen. It would also explain how intermittent the connection is -- your
router is caching DNS request, so they work until they don't, then the page
is update, so the DNS doesn't use the cache, so it works again, but then
the other application requests the port, so it stops working until its
resource is updated or the TTL expires. Pretty vicious cycle!

If you would like me to take a look at your LAN network setting, I may have
time today to pop into the system and check things out, but that's totally
up to you. I can say pretty confidently that a clean installation will
result in the same issues when you try to make the installation public, as
described above.

Yours in support!
Patrick

**Reply by cornerkilling** [reply]
May have spoken too soon on the DuckDNS explanation. I have been running into the same intermittent connectivity issues. I double checked the settings for the NoIP DUC install. I also double checked some settings in my router to make sure there wasn't some weird rule in there. Any help you can offer is welcome.

**Reply by mcguirepr89** [reply]
I have a moment now if you would like to [email me](mailto:mcguirepr89@gmail.com) and I can take a look at your network if you'd like??
